docs/specification/2025-03-26/basic/authorization.md

```diff
@@ -242,14 +241,15 @@ MCP servers integrating with third-party authorization services **MUST**:
 1. Maintain secure mapping between third-party tokens and issued client sessions.  
 2. Validate third-party token status before honoring MCP tokens.  
 3. Implement appropriate token lifecycle management.  
-4. Provide appropriate responses to the client if re-authentication is needed.
+4. Provide appropriate responses to the client if re-authentication is needed. Please refer to
+   [RFC 9470 OAuth 2.0 Step Up Authentication Challenge Protocol](https://www.rfc-editor.org/rfc/rfc9470.html).
 
 #### 2.4.4 Security Considerations
 
 When implementing authorization with identity providers, MCP servers **MUST**:
 
 1. Request the minimum number of required scopes.  
-2. Securely store third-party credentials.  
+2. Securely store any credential artifacts obtained from the client and/or the IdP.  
```

Comment:
Is this still relevant, with the server no longer being an IDP (in most cases)?

Reply:
@PieterKas I more meant the `store` verb. The client is now responsible for storing the tokens (securely), not the server.

Reply:
@nbarbettini - you can still have scenarios where the server _chooses_ to be the broker between the client and itself (e.g., by being the _confidential client_ and helping route the client through the authorization code flow, but it getting the tokens).

Reply:
I actually covered this in a separate section now (`2.5`).